code/modules/projectiles/projectile/energy.dm

```diff
@@ -93,6 +92,58 @@
 	tesla_zap(src, zap_range, power, zap_flags)
 	qdel(src)
 
+/obj/item/projectile/energy/bsg
+	name = "orb of pure bluespace energy"
+	icon_state = "bluespace"
+	impact_effect_type = /obj/effect/temp_visual/bsg_kaboom
+	damage = 60
+	damage_type = BURN
+	range = 9
+	weaken = 1 //This is going to knock you off your feet
+	eyeblur = 5
+	speed = 2
+	alwayslog = TRUE
+
+/obj/item/ammo_casing/energy/bsg/ready_proj(atom/target, mob/living/user, quiet, zone_override = "")
+	..()
+	var/obj/item/projectile/energy/bsg/P = BB
+	spawn(1)
```

Comment:
Does the beam not appear without that?

Reply:
Tested, it is required for the beam to be visable

Reply:
Can you try make this work with `INVOKE_ASYNC`?

I want to shoot `spawn()`